### ISSUE TITLE
Squiz DisallowComparisonAssignmentSniff inconsistent errors with inline IF statements

### ISSUE THREAD

**Issue by AlexHowansky** [squizlabs/PHP_CodeSniffer#616]
I think this sniff has a bug, but I'm not sure what your original intent was. It seems it's supposed to flag things like this:

    $a = $b === true;

... and it does. However, it also flags this:

    $a = $b === true ? $c : $d;

... which seems perfectly fine. Was this intentional?

**Comment by gsherwood** [comment]
You're doing a comparison if both lines of code, and assigning the result to a variable. In the second, you're just assigning the value conditionally. So yes, the sniff should error for both those lines.

The Squiz standard wants you to write your code like this:
```php
if ($b === true) {
    $a = true;
}

if ($b === true) {
    $a = $c;
} else {
    $a = $d;
}
```
But obviously, that's not for everyone. Personally, I find it a lot easier to see what is going on, which is why the Squiz standard enforces it in that way.

**Comment by AlexHowansky** [comment]
Hmm, I still think it's a bug -- or at least an inconsistency. Bear with me here.

First, the `Squiz.PHP.DisallowInlineIf` sniff **already** explicitly covers the inline if operator. As I've excluded that sniff from my ruleset, I should be allowed to use inline ifs -- including the case where I assign its results to a variable.

Second, a comment in `Squiz.PHP.DisallowComparisonAssignment` says, "Ensures that the _value of a comparison_ is not assigned to a variable." In the case of the inline if operator, we're **not** assigning the value of a comparison. I.e., `$b === true` evaluates to the value a of comparison, but `$b === true ? $c : $d` evaluates to the value of a (potentially compound) statement.

Third, with the current code, it is not possible to accomplish my desired standard. I want to allow inline ifs but disallow setting a variable to the value of a comparison. In order to get this behavior, I'd have to exclude your sniff and make my own custom (yet nearly identical) one.

Fourth, and most importantly, if the sniff is really supposed to error for `$a = $b === true ? $c : $d` then it should also error for `$a = $b ? $c : $d`, but it doesn't, which makes no sense.

**Comment by gsherwood** [comment]
> it should also error for $a = $b ? $c : $d, but it doesn't

Then this is a bug. It should error for that as well.

**Comment by gsherwood** [comment]
I've committed a fix so that the errors from this sniff are more predictable. Thanks for pointing out the inconsistency with it.
